fix: UI Styling Issue in Compatibility Percentage Display #6431

chetanr25 opened the pull request:
### What
Adjusted the font size logic for compatibility labels to ensure text visibility across all devices.


### Screenshot

| Before | After |
|---|---|
| ![before](https://github.com/user-attachments/assets/f8514675-bcde-4c00-abc1-583dec7e6a32) | ![after](https://github.com/user-attachments/assets/76ede0f1-5bd7-42cd-b38f-fc9d9fbee710) |
| <img width="399" alt="Screenshot 2025-03-09 at 1 21 04 AM" src="https://github.com/user-attachments/assets/c7caf233-2a45-467a-b0f0-047b902730cd" /> | <img width="397" alt="Screenshot 2025-03-09 at 1 21 36 AM" src="https://github.com/user-attachments/assets/b5abeeb8-8b3d-4f1a-9049-354f8286d513" /> |

I tried modifying EdgeInsets (padding) first, but this resulted in inconsistencies across devices. The font size adjustment provides a reliable solution that maintains proper text visibility on both Android and iOS across various screen sizes and ratios.
A font size of 8.0 for a length of 13 seems to be the sweet spot for all devices.

### Fixes bug(s)
- Fixes: #6431 

### Part of 
- #6431

Commit:
fix: UI Styling Issue in Compatibility Percentage Display #6431

monsieurtanuki reviewed (commented):
Hi @chetanr25!

What would happen with
* `bg`'s "Съвместим" (different character set)
* `cs`'s "Kompatibilní" (2 extra characters)
* `fi`'s "Yhteensopiva" (2 extra characters)

What if you used `AutoSizeText`?
What if the problem was rather the size of the widget, to be computed from the size of the text?

chetanr25 commented:
Now, i checkout with other languages, and this didn't look quite right. The current code had problems with longer text in some languages. I also tried using `AutoSizeText`, but isn't compatible with `IntrinsicHeight`, which we are using above.

So, I wrapped the Text with a `FittedBox` (I kept _getCompatibilityFontSize as well). I tested it with different languages, and now it works fine everywhere. Before, I didn’t check with other languages—thanks for pointing that out!

| Before |  FittedBox | 
|---|---|
| ![before](https://github.com/user-attachments/assets/11fab9eb-465e-41a1-bebb-c1a211b6c8f1) | ![after_fitted_box](https://github.com/user-attachments/assets/ffe55849-371f-42b5-9d1b-f91ac6a70bda) |

| Using  `_getCompatibilityFontSize()` in FittedBox |  Without using it | 
|---|---|
| ![fitted_box_with](https://github.com/user-attachments/assets/e638fe53-88f4-4a01-ab73-dadfb8bc8de9) | ![fitted_box_without](https://github.com/user-attachments/assets/b8474ce0-ca2b-4cd6-bc0c-4ef8c2526616)|

Commit:
Wrapped the text with FittedBox

monsieurtanuki reviewed (commented):
Hi @chetanr25!
Looks like a good improvement indeed!
Please get rid of `_getCompatibilityFontSize` and try not to specify any font size for this `Text` (after all, it's going to be scaled down anyway).

Commit:
Removed _getCompatibilityFontSize from new_product_header file

chetanr25 commented:
Thanks for the feedback! I’ve removed `_getCompatibilityFontSize` and also removed the font size from the Text widget, as suggested. 

| Before removing |  After removing `_getCompatibilityFontSize` | 
|---|---|
| ![before_short](https://github.com/user-attachments/assets/2022518b-b1b0-4f3c-818d-32349c62a136) | ![after_short](https://github.com/user-attachments/assets/f899c7ee-f7b3-483e-bd99-9b6c0ddb2692) | 
|  ![before_long](https://github.com/user-attachments/assets/4ffb3e5f-d77b-4242-8e0b-425191f7b86d) |  ![after_long](https://github.com/user-attachments/assets/b92d4a79-2b40-449e-b83e-73885d55fc61) |

Please have a look at these screenshots and let me know if this works or if I should make any other changes. Thank you!

monsieurtanuki reviewed (approved):
Hi @chetanr25!
I cannot say that I'm 100% sure about the final UI (perhaps some padding needed?), but it's definitely an improvement compared to a partially readable label.
Congratulations on your first contribution here!

chetanr25 commented:
Thank you so much! 
Glad to make my first contribution here. Looking forward to contributing more!
